mirrord/intproxy/src/proxies/outgoing/proxy.rs

```diff
@@ -0,0 +1,165 @@
+use std::collections::HashMap;
+
+use mirrord_protocol::{
+    outgoing::{tcp::DaemonTcpOutgoing, udp::DaemonUdpOutgoing, DaemonConnect, DaemonRead},
+    ConnectionId, RemoteResult,
+};
+
+use crate::{
+    agent_conn::AgentSender,
+    error::{IntProxyError, Result},
+    layer_conn::LayerSender,
+    protocol::{
+        LocalMessage, MessageId, NetProtocol, OutgoingConnectRequest, OutgoingConnectResponse,
+        ProxyToLayerMessage,
+    },
+    proxies::outgoing::interceptor::{InterceptorTask, InterceptorTaskHandle},
+    request_queue::RequestQueue,
+};
+
+pub struct OutgoingProxy {
+    agent_sender: AgentSender,
+    layer_sender: LayerSender,
+    interceptors: HashMap<(ConnectionId, NetProtocol), InterceptorTaskHandle>,
```

Comment:
The key is ConnectionId + NetProtocol? an id alone does not identify an interceptor?

Reply:
I checked this in the agent, it uses separate `ConnectionId` pools for outgoing udp and tcp